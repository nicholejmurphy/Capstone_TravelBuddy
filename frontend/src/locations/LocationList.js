import React from "react";
import LocationCard from "./LocationCard";
import { ButtonGroup, Button } from "reactstrap";

/** Shows a list of all locations
 *
 * - Saves locations as state
 * - Passes 'search()' to LocationSearchForm which allows
 *   usesrs to filter location list by name
 */
function LocationList({ category, setCategory, locations }) {
  return (
    <div className="LocationList container rounded shadow">
      <ButtonGroup>
        <Button
          color="primary"
          outline
          onClick={() => setCategory("geos")}
          active={category === "geos"}
        >
          General
        </Button>
        <Button
          color="primary"
          outline
          onClick={() => setCategory("hotels")}
          active={category === "hotels"}
        >
          Hotels
        </Button>
        <Button
          color="primary"
          outline
          onClick={() => setCategory("restaurants")}
          active={category === "restaurants"}
        >
          Restaurants
        </Button>
        <Button
          color="primary"
          outline
          onClick={() => setCategory("attractions")}
          active={category === "attractions"}
        >
          Attractions
        </Button>
      </ButtonGroup>
      {locations.map((l) => (
        <LocationCard
          key={l.location_id}
          id={l.location_id}
          name={l.location_name}
          address={l.address_obj.address_string}
        />
      ))}
    </div>
  );
}

export default LocationList;
