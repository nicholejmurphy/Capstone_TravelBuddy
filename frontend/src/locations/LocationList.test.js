import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import LocationList from "./LocationList";
import { MemoryRouter } from "react-router";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <LocationList
        locations={[
          {
            location_id: "123",
            name: "test",
            address_obj: { address_string: "test" },
          },
        ]}
      />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LocationList
        locations={[
          {
            location_id: "123",
            name: "test",
            address_obj: { address_string: "test" },
          },
        ]}
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
