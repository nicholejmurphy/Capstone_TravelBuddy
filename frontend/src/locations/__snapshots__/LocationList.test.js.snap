// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LocationList container rounded shadow"
  >
    <div
      class="card"
      style="width: 18rem;"
    >
      <img
        alt="Sample"
        src="https://picsum.photos/300/200"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          test
        </h5>
        <p
          class="card-text"
        >
          test
        </p>
        <button
          class="btn btn-secondary"
          type="button"
        >
          Details
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
